### PR TITLE
USB: netusb: Use standard frame size and cleanup

### LINKED ISSUE #14780: USB: netusb: Unable to send maximum Ethernet packet
USB Ethernet uses `NETUSB_MTU 1500` which does not include ethernet header it is not possible to send maximum Ethernet packet. Affected RNDIS.

### PR DESCRIPTION
**Pull request by finikorg**
Use standard frame size, correct buffer sizes and cleanup.

Fixes #14780

### CODE REVIEW

File: subsys/usb/class/netusb/function_rndis.c [hunk 1]
```diff
@@ -884,7 +867,7 @@ static int handle_encapsulated_cmd(u8_t *data, u32_t len)
 	return 0;
 }
 
-#if SEND_MEDIA_STATUS
+#if IS_ENABLED(SEND_MEDIA_STATUS)
```
Comment (jfischer-phytec-iot) [inline subsys/usb/class/netusb/function_rndis.c]: Could you please add a comment why is it disable per default?

**Comment by finikorg** [comment]
@ubicore Please check this modifications.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: subsys/usb/class/netusb/function_rndis.c (hunk 1 above)
Comment (jukkar) [reply subsys/usb/class/netusb/function_rndis.c]: I think this should be `#if defined(SEND_MEDIA_STATUS)` instead.

Comment (finikorg) [reply subsys/usb/class/netusb/function_rndis.c]: > I think this should be `#if defined(SEND_MEDIA_STATUS)` instead.
I have `SEND_MEDIA_STATUS` defined to `0`.

Comment (finikorg) [reply subsys/usb/class/netusb/function_rndis.c]: > Could you please add a comment why is it disabled per default?

For some reason this option does not work with Windows (8), probably I miss something but it works fine without this media status, probably we do not need it. Maybe I just remove it completely.

**Review by jfischer-no — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**